module/VuFindTheme/src/VuFindTheme/View/Helper/Slot.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ * Slot view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2019.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <challber@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFindTheme\View\Helper;
+
+/**
+ * Slot view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <challber@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Slot extends \Zend\View\Helper\AbstractHelper
+{
+    /**
+     * Storage for strings to be concatinated to the front of a block
+     *
+     * @var array of arrays
+     */
+    protected $blockPrepends = [];
+
+    /**
+     * Storage for strings saved to slots
+     *
+     * @var array
+     */
+    protected $blocks = [];
+
+    /**
+     * Storage for strings to be concatinated to the end of a block
+     *
+     * @var array of arrays
+     */
+    protected $blockAppends = [];
+
+    /**
+     * Call stack to handle nested slots
+     *
+     * @var array
+     */
+    protected $stack = [];
+
+    /**
+     * Get the Slot instance. Create if instance doesn't exist.
+     *
+     * @param string $name Name of target block for action
+     *
+     * @return Slot
+     */
+    public function __invoke($name)
+    {
+        $this->stack[] = $name;
+        return $this;
+    }
+
+    /**
+     * Helper function to return blocks with prepends and appends
+     *
+     * @param string $name Name of target block for action
+     *
+     * @return string|any
+     */
+    protected function build($name)
+    {
+        $pre = $this->blockPrepends[$name] ?? [];
+        $block = $this->blocks[$name] ?? '';
+        $post = $this->blockAppends[$name] ?? [];
+        if (!empty($pre) || !empty($post)) {
+            // TODO: Warn about non-string blocks
```

Comment:
What is the use case for a non-string block?

Reply:
One use case I imagined was lists of links like on our home page or in the footers. For example:

```php
<?php $this->slot('home-links')->set([
  ['Future Events', 'Find a Future Librarian', 'Future Subject Guides'],
  ['Future Databases', 'Future Equipment Rental', 'Board Game Collection']
]); ?>

<?php include $this->parentTemplate('search/home.phtml'); ?>
```

Reply:
I see. Personally, I think it would be more clear to only use slots for actual markup/text, at least within the default templates. We could still allow this possibility in theory, but I think by convention it would be safer not to use it... and if we go that route, I don't think you need to worry about handling the edge case involving non-string blocks and prepend/append.